src/components/App/App.js

```diff
@@ -29,50 +30,90 @@ function App() {
   const [keyword, updateKeyword] = useState("");
   const [isSearchLoading, setIsSearchLoading] = useState(true);
   const [token, setToken] = useState("");
-  const navigate = useNavigate();
-  const handleLogin = async (email, password) => {
-    try {
-      // Call the authorize function with the provided email and password
-      const { token } = await authorize(email, password);
-
-      // Set the token in the state
-      setToken(token);
 
-      // Call the checkToken function to get user data
-      const user = await checkToken(token);
+  const navigate = useNavigate();
+  const handleLogin = (email, password) => {
+    auth
+      .signin(email, password)
+      .then((data) => {
+        if (data.token) {
+          setToken(data.token);
 
-      // Set the current user in the state
-      setCurrentUser(user);
+          auth
+            .checkToken(data.token)
+            .then((res) => {
+              return res;
+            })
+            .then((data) => {
+              setCurrentUser(data);
+            })
+            .then(() => {
+              console.log("logged in");
+              setIsLoggedIn(true);
+            })
+            .then(() => {
+              navigate("/saved-news");
+            })
+            .catch((err) => console.log(err));
 
-      // Set isLoggedIn to true
-      setIsLoggedIn(true);
+          getArticles(data.token).then((data) => {
+            setSavedCards(data);
+          });
+        }
+      })
+      .then(() => {
+        handleClosePopup();
+      })
+      .catch((err) => {
+        console.log(err);
+        setErrorMessage("Username or password is incorrect");
+        setIsLoading(false);
+      });
+  };
 
-      // Navigate to the saved articles page
-      navigate("/saved-news");
+  const handleSuccessPopup = () => {
+    setActivePopup("success");
+  };
 
-      // Fetch articles using the token and set them in the state
-      const articles = await getItems(token);
-      setSavedCards(articles);
+  const handleRegister = (email, password, name) => {
+    console.log(1);
+    console.log(email, password, name);
+    setIsLoading(true);
+    auth
+      .signup(email, password, name)
+      .then((res) => {
+        console.log(4);
+        if (res) {
+          handleSuccessPopup();
+        } else {
+          console.log("Not registered");
+          setErrorMessage("Unsuccessful registeration");
+        }
+      })
 
-      // Close any open popups
-      handleClosePopup();
-    } catch (error) {
-      console.error(error);
-      // Handle authentication error
-      setErrorMessage("Username or password is incorrect");
-      setIsLoading(false);
-    }
+      .then(() => {
+        console.log(5);
+        handleSuccessPopup();
+        setIsLoading(false);
```

Comment:
`setIsLoading(false)` should be in block `finally` not to duplicate in `catch` and in `then`